test/cpp/interop/server_helper_auth.cc

```diff
@@ -0,0 +1,63 @@
+/*
+ *
+ * Copyright 2016, Google Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *     * Neither the name of Google Inc. nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+#include "test/cpp/interop/server_helper.h"
+
+#include <memory>
+
+#include <gflags/gflags.h>
+#include <grpc++/security/server_credentials.h>
+
+#include "src/core/lib/surface/call_test_only.h"
+#include "test/core/end2end/data/ssl_test_data.h"
+
+DECLARE_bool(use_tls);
+
+namespace grpc {
+namespace testing {
+
+std::shared_ptr<ServerCredentials> CreateInteropServerCredentials() {
+  if (FLAGS_use_tls) {
+    SslServerCredentialsOptions::PemKeyCertPair pkcp = {test_server1_key,
+                                                        test_server1_cert};
```

Comment:
Note that I didn't write this. I only moved code from one file to another.

On Mon, Jul 18, 2016 at 11:32 AM, Vijay Pai <notifications@github.com>
wrote:

> In test/cpp/interop/server_helper_auth.cc
> <https://github.com/grpc/grpc/pull/7447#discussion_r71202414>:
>
> > +
> > +#include <gflags/gflags.h>
> > +#include <grpc++/security/server_credentials.h>
> > +
> > +#include "src/core/lib/surface/call_test_only.h"
> > +#include "test/core/end2end/data/ssl_test_data.h"
> > +
> > +DECLARE_bool(use_tls);
> > +
> > +namespace grpc {
> > +namespace testing {
> > +
> > +std::shared_ptr<ServerCredentials> CreateInteropServerCredentials() {
> > +  if (FLAGS_use_tls) {
> > +    SslServerCredentialsOptions::PemKeyCertPair pkcp = {test_server1_key,
> > +                                                        test_server1_cert};
>
> This would normally get my spidey-sense tingling (brace-lists) but this
> case was certainly safe, so it's all good.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/grpc/grpc/pull/7447/files/6b00175833540ad8af4c13627c9690145a9f158b#r71202414>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AG8bptvie7YnjNmN7AR49B2O4lFguKb-ks5qW8bCgaJpZM4JO_6g>
> .
>